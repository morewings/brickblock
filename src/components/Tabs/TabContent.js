import React from 'react';
import PropTypes from 'prop-types';
import {TabContent as Content, TabPane} from 'reactstrap';

const TabContent = props => (
  <Content activeTab={props.activeTab}>
    <TabPane tabId="tab1">
        Tab 1
    </TabPane>
    <TabPane tabId="tab2">
        Tab 2
    </TabPane>
  </Content>
);

TabContent.propTypes = {
  activeTab: PropTypes.string.isRequired,
};

export default TabContent;
