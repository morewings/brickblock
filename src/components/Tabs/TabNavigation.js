import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {Nav, NavItem, NavLink} from 'reactstrap';

const TabNavigation = (props) => {
  const createTabClassName = activeTab => classnames({
    active: activeTab === props.activeTab,
  });
  const onClick = tab => () => {
    props.toggleTab(tab);
  };
  return (
    <Nav tabs>
      <NavItem>
        <NavLink
          className={createTabClassName('tab1')}
          onClick={onClick('tab1')}>
          Tab1
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink
          className={createTabClassName('tab2')}
          onClick={onClick('tab2')}>
          Tab 2
        </NavLink>
      </NavItem>
    </Nav>
  );
};

TabNavigation.propTypes = {
  toggleTab: PropTypes.func.isRequired, // eslint-disable-line
  activeTab: PropTypes.string.isRequired,
};

export default TabNavigation;
