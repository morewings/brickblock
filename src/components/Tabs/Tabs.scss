@import "./../../variables.scss";

.contributors{
  .tab-pane {
    padding-top: $marginVertical;
  }
}

.contributors{
  .nav-tabs .nav-link{
    &.active:hover{
      color: white;
    }
  }
}
