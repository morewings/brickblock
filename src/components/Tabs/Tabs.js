import React, {Fragment} from 'react';
import TabNavigation from 'components/Tabs/TabNavigation';
import TabContent from 'components/Tabs/TabContent';

export default class Tabs extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: 'tab1',
    };
  }
  toggleTab = (activeTab) => {
    this.setState({activeTab});
  };
  render() {
    return (
      <Fragment>
        <TabNavigation toggleTab={this.toggleTab} activeTab={this.state.activeTab} />
        <TabContent activeTab={this.state.activeTab} />
      </Fragment>
    );
  }
}
