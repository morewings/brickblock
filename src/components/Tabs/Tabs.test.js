import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs from 'components/Tabs/Tabs';

describe('src > components > Sample.js', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Tabs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('`this.toggleTab(arg)` changes `state.activeTab === arg`', () => {
    const wrapper = shallow(<Tabs />);
    const sampleTab = 'sampleTab';
    wrapper.instance().toggleTab(sampleTab);
    expect(wrapper.state().activeTab).toEqual(sampleTab);
  });
});
