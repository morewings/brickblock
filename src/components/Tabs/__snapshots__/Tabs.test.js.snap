// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > Sample.js renders without crashing 1`] = `
<React.Fragment>
  <TabNavigation
    activeTab="tab1"
    toggleTab={[Function]}
  />
  <TabContent
    activeTab="tab1"
  />
</React.Fragment>
`;
